setup.py

```diff
@@ -356,5 +407,6 @@ def run(self):
           cmdclass={'clean': Clean,
                     'develop': Develop,
                     'lint': Linter,
```

Comment:
Will this `lint` target be away, since we have the `static-checks` now?

Reply:
This is something we need to discuss. There is still a difference between both options, `python3 setup.py lint` works without installing avocado while `python3 setup.py selftests --static-checks` needs avocado installed (in develop mode) and uses avocado to run the tests.

Reply:
I see. Well I don't see a reason to have both here. And since avocado is already used by others checks, I'm not sure if we need the first one. And TBH, unless we have a strong argument, having mixing checks were some uses avocado and others don´t, seems confusing. So let's see what others think about this.

Reply:
IMO the selftest checks is for the contributors whose have to work in develop mode anyway, so I agree with @beraldoleal  that we do not need the `python3 setup.py lint` check

Reply:
@ana have you considered using the `test` command instead?  I mean, Python developers will expect to be able to run `python setup.py test`, so replacing it, instead of adding `selftests` seems like the best option here.

Reply:
> @ana have you considered using the `test` command instead? I mean, Python developers will expect to be able to run `python setup.py test`, so replacing it, instead of adding `selftests` seems like the best option here.

yes, I discarded it because `python setup.py test` already does something (running unit test).
The command is deprecated:  you get a message **"WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox"**

If you prefer it, I'm happy to override the actual command and use `test ` instead of `selftests`

Reply:
> > @ana have you considered using the `test` command instead? I mean, Python developers will expect to be able to run `python setup.py test`, so replacing it, instead of adding `selftests` seems like the best option here.
> 
> yes, I discarded it because `python setup.py test` already does something (running unit test).
> The command is deprecated: you get a message **"WARNING: Testing via this command is deprecated and will be removed in a future version. Users looking for a generic test entry point independent of test runner are encouraged to use tox"**
> 

OK... still, it's not clear to me if they are deprecating the *built-in runner* or the command/interface.

> If you prefer it, I'm happy to override the actual command and use `test ` instead of `selftests`

I guess my final preference will depend on the previous point... do you think we can find that out?

Reply:
> OK... still, it's not clear to me if they are deprecating the _built-in runner_ or the command/interface.
> 
> > If you prefer it, I'm happy to override the actual command and use `test ` instead of `selftests`
> 
> I guess my final preference will depend on the previous point... do you think we can find that out?

The setuptools command is being deprecated. More information at https://github.com/pypa/setuptools/issues/1684

Reply:
IIUC, we mixed the topics here. My initial comment it was about 'lint' command. And by @richtja comment and @clebergnu thumbs up, I'm assuming that we could drop it. Did I understood it right?

Reply:
> IIUC, we mixed the topics here. My initial comment it was about 'lint' command. And by @richtja comment and @clebergnu thumbs up, I'm assuming that we could drop it. Did I understood it right?

Yeah, we discussed two different things. I addressed both:
- `lint` has been deprecated (instead of removing it, now it'll show a message telling you to use `python setup.py test --static-check`)
- `selftests` has been renamed to `test`.